### DISCUSSION TITLE
Partitioning/file system considerations when hosting virtual machines?

### DISCUSSION THREAD

**Discussion by MartinHemma** [SpiralLinux/SpiralLinux-project#365] [Q&A] [answered]
New computer delivered. Spiral Linux (XFCE) is strongly considered as I already run that on my 'side-kick' laptop and like it. The laptop install is very basic. Defaults + full disk encryption.

The new computer will replace my 13 year old main machine. This is where I will keep/edit photos and to a lesser degree videos. This is where I tinker with Python, keep my financial data etc. This is the machine I back up on a regular basis.

This is also the machine that will host a handful of virtual machines -- both Virtualbox- and Qemu-based.

The old machine has two spinning disk HDDs. The new machine have one SSD (for now).

On the old machine the virtual machines live on a separate, ext4 partition while / and /home are btrfs partitions. I created this setup back in 2016 and don't remember why.

Is there any reason for why I shouldn't have my virtual machines in a folder under /home (assuming btrfs)?

Any other SSD considerations during install? Separate partitions for / and /home or not? Btrfs vs ext4? Swap?

TIA

/Martin

**Comment by MartinHemma** [comment]
I have done some thinking since yesterday:

Btrfs is a COW (Copy On Write) file system.
The files that are 'disks' for virtual machines tend to be large. My two largest are 45 and 20+ GB. I guess COW can be challenging for such large files. At least for 2010 'home office' HW.

Does this make sense?

Does it still make sense in 2023? The new computer has 32 GB RAM and 1 TB SSD.

/Martin

**Reply by AlphaElwedritsch** [reply]
`Does it still make sense in 2023? The new computer has 32 GB RAM and 1 TB SSD.`
It depends on the use case

**Reply by MartinHemma** [reply]
Use case(s) for the computer or for the virtual machines?

The virtual machines have two use cases:

1. Testing before implementing on real machine.
2. Access to SW only available on Windows.

The first one is kind of transient use. Those machines don't live long.

The second one is occasional use and those machines have been around for many years. Those machines have no internet access so I am not worried about running rather old Windows version there.

There is a third use case. I have CAELinux installed on a machine with the intent to try out FOSS FEM SW. This is the 45 GB machine. I hope I can retire this one after (if) I install that SW on my new, real machine. I am currently watching this: http://tuxriders.com/videos/setup/

Late yesterday I came across a btrfs vs ext4 comparison in which the author claimed btrfs is better at handling large files. It was not a deep dive to put it mildly.

/Martin

**Reply by geckolinux** [reply]
The huge advantage to Btrfs in my opinion is that it uses checksums to verify the integrity of all files. This reduces the risk of hidden corruption silently eating your data.

I personally use a Windows VM on top of a Btrfs filesystem and it works fine.

If you don't want the automatic Snapper snapshot routines to snapshot your VMs you can convert the VM directory to a Btrfs subvolume similar to this suggestion:
https://www.reddit.com/r/btrfs/comments/fi36l8/how_can_i_prevent_snapshots_on_a_directory/fkewusg/

**Reply by MartinHemma** [reply]
Thanks for this response. I had already 'gravitated' towards the decision to go for the 'default' btrfs install and host my VMs in a folder rather than on a separate partition. Your response supports this.

I will study the material you linked.

/Martin

**Comment by MartinHemma** [comment]
Leaning more and more towards a default btrfs install and keeping virtual machines in a folder under /home. If that does not work well I will simply have to reconsider.

@geckolinux Today I noticed F2FS is an option during installation. Worth considering? I had not heard of F2FS before so btrfs is much more familiar. Also, I know you set up btrfs in a, from my point of view, clever way but know nothing about what you do with F2FS.

/Martin

**Reply by geckolinux** [reply, marked as answer]
Hi there, in SpiralLinux the only filesystem that comes with a specific and unique configuration is Btrfs. F2FS doesn't have any fancy user-facing features, but at a low level it's optimized for flash storage.
